back-end/src/sciler/config/workingConfigTypes.go

```diff
@@ -0,0 +1,320 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// WorkingConfig has additional fields to ReadConfig, with lists of conditions, constraints and actions.
+type WorkingConfig struct {
+	General       General
+	Puzzles       []Puzzle
+	GeneralEvents []GeneralEvent
+	Devices       map[string]Device
+}
+
+// Rule is a struct that describes how action flow is handled in the escape room.
+type Rule struct {
+	ID          string
+	Description string
+	Limit       int
+	Conditions  LogicalCondition
+	Actions     []Action
+}
+
+// Puzzle is a struct that describes contents of a puzzle.
+type Puzzle struct {
+	Event GeneralEvent
+	Hints []string
+}
+
+// GetName returns the name of a Puzzle
+func (p Puzzle) GetName() string {
+	return p.Event.Name
+}
+
+// GetRules returns the rules of a Puzzle
+func (p Puzzle) GetRules() []Rule {
+	return p.Event.Rules
+}
+
+// GeneralEvent defines a general event, like start.
+type GeneralEvent struct {
```

Comment:
why is this defined again? why not use the one from reading config as there is no difference?

Reply:
They are different